motr/client_init.c

```diff
@@ -1250,8 +1268,25 @@ static int initlift_idx_service(struct m0_sm *mach)
 		rc = service->is_svc_ops->iso_init((void *)ctx);
 		m0_sm_group_lock(&m0c->m0c_sm_group);
 
-		if (rc != 0)
-			initlift_fail(rc, m0c);
+		if (rc != 0) {
+			/*
+			 * Added retry logic to handle out of
+			 * order startup of data and server
+			 * PODs. Ref: Jira ID Cortx-33899
+			 */
+			if (retry_count < MAX_CLIENT_INIT_RETRIES
+			    && (rc == -EIO || rc == -EPROTO)) {
+				retry_count += 1;
```

Comment:
Do we need to add some sleep()/delay() here?

Reply:
Do we have some kind of pod/service dependencies?
For example, the "server pods" can only be started after the "data pods" are successfully started.

Reply:
Added sleep in my draft, but madhav wants to avoid it. @madhavemuri please share your feedback
yes,  server pod has a dependency on data pod.

Reply:
During first deployment all the server pods wait until data pods run time containers are started.
I think this issue issue is seen most when cluster restart is done.

Reply:
Yes, the issue is seen with cluster restart only.

Reply:
Introduced sleep between retires in exponential backoff manner. Now  m0_client_init() will retry IDX service initialization for around 4min in total,  and return error post that.